File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/tinsel/TitlesAndOrnamentsListMessage.scala





















// Generated on 05/03/2013 10:35:00
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait TitlesAndOrnamentsListMessage extends MessageDeserializer {
  val messageId = 6367

  protected trait Message extends BaseMessage with Serializable {
    val messageId = TitlesAndOrnamentsListMessage.messageId

    val titles: Seq[Short]
    val ornaments: Seq[Short]
    val activeTitle: Short
    val activeOrnament: Short

    def serialize(buf: Buffer) {
      buf.writeUShort(titles.length)
      for (entry <- titles) {
        buf.writeShort(entry)
      }
      buf.writeUShort(ornaments.length)
      for (entry <- ornaments) {
        buf.writeShort(entry)
      }
      buf.writeShort(activeTitle)
      buf.writeShort(activeOrnament)
    }

  }

  protected def create(titles: Seq[Short], ornaments: Seq[Short], activeTitle: Short, activeOrnament: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val titles = {
      val builder = collection.immutable.Seq.newBuilder[Short]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readShort()
      }
      builder.result()
    }
    val ornaments = {
      val builder = collection.immutable.Seq.newBuilder[Short]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readShort()
      }
      builder.result()
    }
    val activeTitle = buf.readShort()
    val activeOrnament = buf.readShort()

    create(titles, ornaments, activeTitle, activeOrnament)
  }

}

object TitlesAndOrnamentsListMessage extends TitlesAndOrnamentsListMessage {
  case class MessageImpl(titles: Seq[Short], ornaments: Seq[Short], activeTitle: Short, activeOrnament: Short) extends Message
  type Target = MessageImpl

  override def create(titles: Seq[Short], ornaments: Seq[Short], activeTitle: Short, activeOrnament: Short) = MessageImpl(titles, ornaments, activeTitle, activeOrnament)

  def apply(titles: Seq[Short], ornaments: Seq[Short], activeTitle: Short, activeOrnament: Short) = create(titles, ornaments, activeTitle, activeOrnament)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/TitlesAndOrnamentsListMessage.java


// Generated on 05/08/2013 19:37:58
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class TitlesAndOrnamentsListMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6367;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short[] titles;
    public short[] ornaments;
    public short activeTitle;
    public short activeOrnament;
    
    public TitlesAndOrnamentsListMessage() { }
    
    public TitlesAndOrnamentsListMessage(short[] titles, short[] ornaments, short activeTitle, short activeOrnament) {
        this.titles = titles;
        this.ornaments = ornaments;
        this.activeTitle = activeTitle;
        this.activeOrnament = activeOrnament;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeUShort(titles.length);
        for (short entry : titles) {
            buf.writeShort(entry);
        }
        buf.writeUShort(ornaments.length);
        for (short entry : ornaments) {
            buf.writeShort(entry);
        }
        buf.writeShort(activeTitle);
        buf.writeShort(activeOrnament);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        int limit = buf.readUShort();
        titles = new short[limit];
        for (int i = 0; i < limit; i++) {
            titles[i] = buf.readShort();
        }
        limit = buf.readUShort();
        ornaments = new short[limit];
        for (int i = 0; i < limit; i++) {
            ornaments[i] = buf.readShort();
        }
        activeTitle = buf.readShort();
        if (activeTitle < 0)
            throw new RuntimeException("Forbidden value on activeTitle = " + activeTitle + ", it doesn't respect the following condition : activeTitle < 0");
        activeOrnament = buf.readShort();
        if (activeOrnament < 0)
            throw new RuntimeException("Forbidden value on activeOrnament = " + activeOrnament + ", it doesn't respect the following condition : activeOrnament < 0");
    }
    
}
